packages/core/content-manager/server/services/utils/populate/index.js

```diff
@@ -191,7 +213,72 @@ const getDeepPopulateDraftCount = (uid) => {
   return { populate, hasRelations };
 };
 
+/**
+ *  Create a Strapi populate object which populates all attribute fields of a Strapi query.
+ *
+ * @param {string} uid
+ * @param {Object} query
+ * @returns {Object} populate object
+ */
+const getQueryPopulate = async (uid, query) => {
+  const pathToPopulate = (path) => path.replaceAll('.', '.populate.');
+  let populateQuery = {};
+
+  await strapiUtils.traverse.traverseQueryFilters(
+    /**
+     *
+     * @param {Object} param0
+     * @param {string} param0.key - Attribute name
+     * @param {Object} param0.attribute - Attribute definition
+     * @param {string} param0.path - Content Type path to the attribute
+     * @returns
+     */
+    ({ key, attribute, path }) => {
+      // Ignore dynamic zones and morph relations
+      if (!attribute || isDynamicZone(attribute) || isMorphRelation(attribute)) {
```

Comment:
we might also want to include dynamic zones, how would that work with custom conditions though? Same for polymorphic. Do we support custom conditions that work for dynamic zones or polymorphic relations.

Reply:
I agree this method could support it but if it's not really needed maybe we could add it as a TODO 🤔

Reply:
Yes let's add it as a todo, we'll need to investigate that a bit more in the future

Reply:
What about https://github.com/strapi/strapi/pull/16814#discussion_r1206724940 though?

Reply:
you are right , we could keep the reference side! omw

Reply:
btw, I was checking the definition of morph relations because I always forget 😄 https://github.com/strapi/strapi-v4-rfc/blob/master/docs/database/relations.md#polymorphic-relations

Shouldn't we pick the side which always point to the same content type? by looking at the diagrams is not the morphTo but morph

Reply:
Yes exactly, that's why I was talking about omitting the morphTo ones :slightly_smiling_face:

Reply:
done! :)